lib/git_service/branch.rb

```diff
@@ -53,8 +53,24 @@ def target_for_reference(reference) # Rugged::Commit for a given refname i.e. "r
       rugged_repo.references[reference].target
     end
 
+    def tip_files
+      recursive_list_files_in_tree(tip_tree.oid)
+    end
+
     private
 
+    def recursive_list_files_in_tree(rugged_tree_oid, files = [], current_path = Pathname.new(""))
+      rugged_repo.lookup(rugged_tree_oid).each do |i|
+        case i[:type]
+        when :blob
+          files << current_path.join(i[:name]).to_s
```

Comment:
You can extract `current_path.join(i[:name])` to a variable since it's in both paths